### PR TITLE
OpenJK: init at 2017-08-11

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Community effort to maintain and improve Star Wars Jedi Academy + Jedi Outcast game engines released by Raven Software

https://github.com/JACoders/OpenJK

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/games/openjk/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  dontAddPrefix = true;
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [ makeWrapper cmake ];
+  buildInputs = [ libjpeg zlib libpng mesa_noglu SDL2 ];
+
+  preConfigure = ''
+    cmakeFlags="-DCMAKE_INSTALL_PREFIX=$out/opt"
```
Comment (Mic92) [inline pkgs/games/openjk/default.nix]: this should be `cmakeFlagsArray` in bash.

Comment (gnidorah) [inline pkgs/games/openjk/default.nix]: @Mic92 Fixed, thanks! BTW do you know a solution for this issue? https://github.com/NixOS/nixpkgs/pull/27096#issuecomment-324060858

File: pkgs/games/openjk/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  };
+
+  dontAddPrefix = true;
+  enableParallelBuilding = true;
+
+  nativeBuildInputs = [ makeWrapper cmake ];
+  buildInputs = [ libjpeg zlib libpng mesa_noglu SDL2 ];
+
+  preConfigure = ''
+    cmakeFlagsArray=("-DCMAKE_INSTALL_PREFIX=$out/opt")
```
Comment (Mic92) [inline pkgs/games/openjk/default.nix]: Any reason not to use `"-DCMAKE_INSTALL_PREFIX=$out"` instead?

Comment (gnidorah) [inline pkgs/games/openjk/default.nix]: @Mic92 It then installs engine under `$out/JediAcademy`. Maybe put a comment about it?

Comment (gnidorah) [inline pkgs/games/openjk/default.nix]: @Mic92 Ok, I've put a comment why we need it.

**Pull request merged**